### DISCUSSION TITLE
Enable touch events

### DISCUSSION THREAD

**Discussion by Reiji777** [CoderLine/alphaTab#641] [Ideas]
As I tested the example on the homepage and took a quick look at the source code, it seems for now only the mouse events `mousedown`, `mouseup`, `mousemove` are available. I was wondering what would be a good way to support touch events too?

I'm asking mainly because I'd like to trigger `cursorSelectRange()` via `touchmove`, just like `mousemove`.

**Comment by Danielku15** [comment]
As you have found out already, touch interaction is not supported currently. In general we cannot/should not hook up the touch events directly because this will rather cause troubles for most users. When you want to just scroll through the tabs, you don't want to seek or select anything. 


It's hard to say if there is a good solution on touch interaction that fits all needs. There are many approaches on touch devices. One way regarding selection could be a long-press to activate the selection mode. For seeking also a long press could do the job, or rather a real single-tap gesture. 

The main magic regarding user interaction is hooked up here. There is not much hidden logic that wouldn't be exposed. 
https://github.com/CoderLine/alphaTab/blob/f1f2af50534dd5ae69848067716622f9457c7e90/src/AlphaTabApiBase.ts#L1082

I am happy to accept contributions in regards to touch interaction, but for now it is not really my focus. If anybody works out a proper solution we could integrate it to alphaTab itself.
